Framework/MDAlgorithms/src/ConvertCWPDMDToSpectra.cpp

```diff
@@ -287,15 +290,8 @@ API::MatrixWorkspace_sptr ConvertCWPDMDToSpectra::reducePowderData(
     pdws->getAxis(0)->setUnit("Degrees");
   }
 
-  MantidVec &dataX = pdws->dataX(0);
-  for (size_t i = 0; i < sizex; ++i)
-    dataX[i] = vecx[i];
-  MantidVec &dataY = pdws->dataY(0);
-  MantidVec &dataE = pdws->dataE(0);
-  for (size_t i = 0; i < sizey; ++i) {
-    dataY[i] = vecy[i];
-    dataE[i] = vece[i];
-  }
+  pdws->setHistogram(0, BinEdges(vecx), Counts(vecy),
```

Comment:
This seems like a significant improvement, have you noticed any difference in the tests execution time? (If they are long enough to be noticeable)